### PR TITLE
Feature - PIL review

### PR DESCRIPTION
**Pull request by joefitter**
* Added pil review endpoint
* Updated permissions middleware to take an array of allowed permissions

### CODE REVIEW

File: lib/routers/profile/pil.js [hunk 1]
```diff
@@ -134,19 +134,26 @@ router.post('/',
...
 
 router.put('/:pilId/:action',
-  permissions('pil.update'),
   checkEstablishment,
   validateAction,
   validate
 );
 
+router.put('/:pilId/review',
+  permissions(['pil.update', 'pil.review']),
```
Comment (lennym) [inline lib/routers/profile/pil.js]: Does this require both, or just one?

File: lib/middleware/permissions.js [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
+  tasks = castArray(tasks);
+  if (req.permissionHoles && !without(tasks, ...req.permissionHoles).length) {
     return next();
   }
-  req.user.can(task, { ...req.params, ...params(req, res) })
-    .then(can => can ? next() : next(new UnauthorisedError()))
+  Promise.all(
+    tasks.map(task => req.user.can(task, { ...req.params, ...params(req, res) }))
+  )
+    .then(cans => some(cans) ? next() : next(new UnauthorisedError()))
```
Comment (lennym) [inline lib/middleware/permissions.js]: I think this is the answer to my question. It only requires one permission to be good?

File: lib/routers/profile/pil.js (hunk 1 above)
Comment (lennym) [reply lib/routers/profile/pil.js]:
```suggestion
  // user must have update OR review permissions
  permissions(['pil.update', 'pil.review']),
```

Comment (lennym) [reply lib/routers/profile/pil.js]: So it's obvious without finding the downstream code.

File: lib/middleware/permissions.js (hunk 2 above)
Comment (joefitter) [reply lib/middleware/permissions.js]: yeah - it's because an NTCO cannot update, but can review, so you either need update or review permissions

**Pull request merged**